src/alchemlyb/convergence/convergence.py

```diff
@@ -137,3 +138,171 @@ def forward_backward_convergence(df_list, estimator='MBAR', num=10, **kwargs):
          'data_fraction': [i / num for i in range(1, num + 1)]})
     convergence.attrs = df_list[0].attrs
     return convergence
+
+def _cummean(vals, out_length):
+    '''The cumulative mean of an array.
+
+    This function computes the cumulative mean and shapes the result to the
+    desired length.
+
+    Parameters
+    ----------
+    vals : numpy.array
+        The one-dimensional input array.
+    out_length : int
+        The length of the output array.
+
+    Returns
+    -------
+    numpy.array
+        The one-dimensional input array with length of total.
+
+    Note
+    ----
+    If the length of the input series is smaller than the ``out_length``, the
+    length of the output array is the same as the input series.
+
+
+    .. versionadded:: 1.0.0
+
+    '''
+    in_length = len(vals)
+    if in_length < out_length:
+        out_length = in_length
+    block = in_length // out_length
+    reshape = vals[: block*out_length].reshape(block, out_length)
+    mean = np.mean(reshape, axis=0)
+    result = np.cumsum(mean) / np.arange(1, out_length+1)
+    return result
+
+def fwdrev_cumavg_Rc(series, precision=0.01, tol=2):
+    '''Generate the convergence criteria :math:`R_c` for a single simulation.
+
+    The input will be pandas.Series generated by
+    :func:`~alchemlyb.preprocessing.subsampling.decorrelate_u_nk` or
+    :func:`~alchemlyb.preprocessing.subsampling.decorrelate_dhdl`.
+
+    The output will the float :math:`R_c`. :math:`R_c = 0` indicates that the system is well
+    equilibrated right from the beginning while :math:`R_c = 1` signifies that the
+    whole trajectory is not equilibrated.
+
+    Parameters
+    ----------
+    series : pandas.Series
+        The input energy array.
+    precision : float
+        The precision of the output :math:`R_c`. To speed the calculation up, the data
+        has been block-averaged before doing the calculation, the size of the
+        block is controlled by the desired precision.
+    tol : float
+        Tolerance of the convergence check in :math:`kT`.
+
+    Returns
+    -------
+    float
+        Convergence time fraction.
+    DataFrame
+        The DataFrame with moving average. ::
+
+                Forward  Backward  data_fraction
+            0  3.016442  3.065176            0.1
+            1  3.078106  3.078567            0.2
+            2  3.072561  3.047357            0.3
+            3  3.048325  3.057527            0.4
+            4  3.049769  3.037454            0.5
+            5  3.034078  3.040484            0.6
+            6  3.043274  3.032495            0.7
+            7  3.035460  3.036670            0.8
+            8  3.042032  3.046597            0.9
+            9  3.044149  3.044385            1.0
+
+    Note
+    ----
+    This function computes :math:`R_c` from equation 16 from
+    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8397498/#FD16. The code is
+    modified based on Shujie Fan's (@VOD555) work.
+
```

Comment:
```suggestion
   
   Please cite [Fan2021]_ when using this function.
   
```
Can you check that the reference resolves?